oak_functions/abi/src/lib.rs

```diff
@@ -47,12 +47,12 @@ pub mod proto {
         }
     }
 
-    // Converts a ChannelStatus into an OakStatus.
-    impl From<OakStatus> for ChannelStatus {
-        fn from(oak_status: OakStatus) -> Self {
-            match oak_status {
-                OakStatus::ErrInvalidArgs => ChannelStatus::ChannelInvalidArgs,
-                _ => ChannelStatus::Unspecified,
+    // Converts a OakStatus into an ChannelStatus.
+    impl From<ChannelStatus> for OakStatus {
+        fn from(channel_status: ChannelStatus) -> Self {
+            match channel_status {
+                ChannelStatus::ChannelInvalidArgs => OakStatus::ErrInvalidArgs,
+                _ => OakStatus::Unspecified,
```

Comment:
It seems like it might be an issue in future if there is not a channel status related to and internal error if all statusses will be channel statusses in future.

Reply:
Good point about error codes being transmitted over the channel, rather than being indicated by the channel status. I guess the current wierd state is because the transition is in-progress.